### PR TITLE
[ASCII-1985] Allow to fetch config check information and do not scrub the configuration information

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Add a the `raw` URL param to the `config-check` endpoint. Using the URL param allow us to teel the server if the configuration should be scrubbed or not.

The SNMP walk command needs the community strings to not being scrubbed so they can perform an snmpwalk like calls.

The regression was introduced in this PR https://github.com/DataDog/datadog-agent/pull/27389

### Describe how to test/QA your changes

- Ensure CLI command agent `snmp walk` works as expected.

### CODE REVIEW

File: comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go [hunk 1]
```diff
@@ -195,8 +195,15 @@ func (ac *AutoConfig) serviceListening() {
 	}
 }
 
-func (ac *AutoConfig) writeConfigCheck(w http.ResponseWriter, _ *http.Request) {
-	configCheckResponse := ac.GetConfigCheck()
+func (ac *AutoConfig) writeConfigCheck(w http.ResponseWriter, r *http.Request) {
+	raw := r.URL.Query().Get("raw") == "true"
```
Comment (davidor) [inline comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go:199]: r is a pointer. Should we add a nil check?

Comment (GustavoCaso) [reply comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go:199]: The `writeConfigCheck` function is called by the HTTP mux 

https://github.com/DataDog/datadog-agent/blob/ac6df4bac214e2ff6a242c6315418363f34abfe5/comp/api/api/apiimpl/internal/agent/agent.go#L53

I doubt that `r` would be nil, but adding the extra check would not hurt

Comment (GustavoCaso) [reply comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go]: Done https://github.com/DataDog/datadog-agent/pull/27568/commits/af72f5f1af8caa20c6fd1a8d96c48e4844dab594

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

**Review by AlexandreYang — APPROVED** [review]

**Review by TCheruy — APPROVED** [review]

**Pull request merged**